ConsumerApi/SynchronizationDbContextSeeder.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Text.Json;
+using Backbone.BuildingBlocks.API.Extensions;
+using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.BlobStorage;
+using Backbone.BuildingBlocks.Application.Extensions;
+using Backbone.BuildingBlocks.Application.Pagination;
+using Backbone.Modules.Synchronization.Application.Infrastructure;
+using Backbone.Modules.Synchronization.Domain.Entities;
+using Backbone.Modules.Synchronization.Infrastructure.Persistence.Database;
+using Backbone.Tooling.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Backbone.ConsumerApi;
+
+public class SynchronizationDbContextSeeder : IDbSeeder<SynchronizationDbContext>
+{
+    private readonly IBlobStorage? _blobStorage;
+    private readonly string? _blobRootFolder;
+
+    public SynchronizationDbContextSeeder(IServiceProvider serviceProvider)
+    {
+        _blobStorage = serviceProvider.GetService<IBlobStorage>();
+        _blobRootFolder = serviceProvider.GetService<IOptions<BlobOptions>>()!.Value.RootFolder;
+    }
+
+    public async Task SeedAsync(SynchronizationDbContext context)
+    {
+        await FillEncryptedPayloadColumnFromBlobStorage(context);
+    }
+
+    private async Task FillEncryptedPayloadColumnFromBlobStorage(SynchronizationDbContext context)
+    {
+        // _blobRootFolder is null when blob storage configuration is not provided, meaning the content of database entries should not be loaded from blob storage
+        if (_blobRootFolder == null)
+            return;
+
+        var paginationResult = await context.DatawalletModifications
+            .Where(t => t.EncryptedPayload == null)
+            .OrderAndPaginate(m => m.Id, new PaginationFilter(1, 20), CancellationToken.None);
```

Comment:
Now you're just migrating the first 20 items... 😑 

Also, you can increase the page size to at least 500.

Reply:
Ok, that's what I was afraid of. I'll get it immediately.